### PR TITLE
HV Boot and Wheels Spinning

### PR DESCRIPTION
**Pull request by Lucien950**
### Changelist 

- Make wheels turn

### Testing Done

- Turning wheels

### Resolved Tickets

uh

### CODE REVIEW

File: firmware/quadruna/VC/src/io/io_tsms.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+typedef struct
+{
+    const Gpio *tsms_gpio;
+} TSMSConfig;
+#else
+#include "app_utils.h"
+EMPTY_STRUCT(PcmConfig)
+#endif
+
+void io_tsms_init(const TSMSConfig *const in_config);
```
Comment (will-chaba) [inline firmware/quadruna/VC/src/io/io_tsms.h:12]: missing function descriptions

File: firmware/quadruna/VC/src/tasks.c [hunk 2]
```diff
@@ -349,14 +356,17 @@ void tasks_init(void)
...
     io_currentSensing_init(&current_sensing_config);
     io_efuse_init(efuse_configs);
     app_efuse_init(efuse_enabled_can_setters, efuse_current_can_setters);
 
     app_canTx_VC_Hash_set(GIT_COMMIT_HASH);
     app_canTx_VC_Clean_set(GIT_COMMIT_CLEAN);
+
+    // TODO enable these for inverter programming
+    //    hw_gpio_writePin(&inv_l_program, true);
+    //    hw_gpio_writePin(&inv_r_program, true);
```
Comment (will-chaba) [inline firmware/quadruna/VC/src/tasks.c:369]: Just commenting to highlight this for later

File: firmware/quadruna/VC/src/tasks.c [hunk 3]
```diff
@@ -403,7 +412,6 @@ _Noreturn void tasks_run100Hz(void)
     for (;;)
     {
         //        const uint32_t start_time_ms = osKernelGetTickCount();
```
Comment (will-chaba) [inline firmware/quadruna/VC/src/tasks.c:414]: Don't think this is needed 
```suggestion
```

File: firmware/quadruna/VC/src/tasks.c [hunk 4]
```diff
@@ -468,7 +476,7 @@ _Noreturn void tasks_runCanRx(void)
     {
         CanMsg rx_msg;
         io_can_popRxMsgFromQueue(&rx_msg);
-        io_telemMessage_broadcast(&rx_msg);
+        //        io_telemMessage_broadcast(&rx_msg);
```
Comment (will-chaba) [inline firmware/quadruna/VC/src/tasks.c:479]: is this desired?

File: firmware/shared/src/app/app_heartbeatMonitor.h [hunk 5]
```diff
@@ -1,55 +1,35 @@
+// TODO add grace period for heartbeat monitor checking
```
Comment (will-chaba) [inline firmware/shared/src/app/app_heartbeatMonitor.h:1]: bringing attention to TODO

File: firmware/shared/test/test_heartbeatMonitor.cpp [hunk 6]
```diff
@@ -239,20 +238,20 @@ TEST(HeartbeatMonitor, test_check_in_and_tick)
     HeartbeatMonitor *heartbeat_monitor = app_heartbeatMonitor_get();
 
     // assert nothing changed (0 ms)
-    app_heartbeatMonitor_tick();
-    ASSERT_EQ(heartbeat_monitor->previous_timeout_ms, 0U);
+    app_heartbeatMonitor_checkIn();
+    //    ASSERT_EQ(heartbeat_monitor->previous_timeout_ms, 0U);
```
Comment (will-chaba) [inline firmware/shared/test/test_heartbeatMonitor.cpp:242]: Update or remove

File: firmware/shared/test/test_heartbeatMonitor.cpp [hunk 7]
```diff
@@ -279,9 +278,9 @@ TEST(HeartbeatMonitor, test_check_in_and_tick)
 
     // progress to timeout period and verify all good (300 ms)
     fake_io_time_getCurrentMs_returns(300);
-    app_heartbeatMonitor_tick();
+    app_heartbeatMonitor_checkIn();
 
-    ASSERT_EQ(heartbeat_monitor->previous_timeout_ms, 300);
+    //    ASSERT_EQ(heartbeat_monitor->previous_timeout_ms, 300);
```
Comment (will-chaba) [inline firmware/shared/test/test_heartbeatMonitor.cpp:283]: update or remove

File: firmware/shared/test/test_heartbeatMonitor.cpp [hunk 8]
```diff
@@ -313,9 +312,9 @@ TEST(HeartbeatMonitor, test_check_in_and_tick)
 
     // progress to timeout period and verify A missing (600 ms)
     fake_io_time_getCurrentMs_returns(600);
-    app_heartbeatMonitor_tick();
+    app_heartbeatMonitor_checkIn();
 
-    ASSERT_EQ(heartbeat_monitor->previous_timeout_ms, 600);
+    //    ASSERT_EQ(heartbeat_monitor->previous_timeout_ms, 600);
```
Comment (will-chaba) [inline firmware/shared/test/test_heartbeatMonitor.cpp:317]: update or remove

**Review by will-chaba — CHANGES_REQUESTED** [review]
Bunch of small changes and comments highlighting TODO sections. Big PR so want to make sure any "temporary" debugging features get removed

File: can_bus/quadruna/BMS/BMS_tx.json [hunk 9]
```diff
@@ -401,5 +401,18 @@
                 "bits": 1
             }
         }
+    },
+    "ModuleCommunication": {
```
[the remark this answers is not in the document]
Comment (Lucien950) [reply can_bus/quadruna/BMS/BMS_tx.json:405]: I think still good to have for the long term?

File: firmware/quadruna/BMS/src/app/states/app_allStates.c [hunk 10]
```diff
@@ -53,7 +53,6 @@ bool app_allStates_runOnTick100Hz(void)
     app_canTx_BMS_Heartbeat_set(true);
 
     app_heartbeatMonitor_checkIn();
-    app_heartbeatMonitor_tick();
```
[the remark this answers is not in the document]
Comment (Lucien950) [reply firmware/quadruna/BMS/src/app/states/app_allStates.c:56]: yes

File: firmware/quadruna/VC/src/tasks.c (hunk 4 above)
Comment (Lucien950) [reply firmware/quadruna/VC/src/tasks.c:479]: yes. Telem messages are clogging the rxtask on vc

**Comment by Lucien950** [comment]
ok @will-chaba changes addressed.

**Comment by Lucien950** [comment]
I wanna merge in today

**Comment by will-chaba** [comment]
> I wanna merge in today

Respectfully, should not be merged until driving shakedown

**Comment by Lucien950** [comment]
> > I wanna merge in today
> 
> Respectfully, should not be merged until driving shakedown

was worried about rebasing especially with the things we want (like telem + sd card)

**Review by will-chaba — APPROVED** [review]
LGTM! 🚗

**Pull request merged**